From cc90d4a563c070a58dadba7f0e98f252a080d225 Mon Sep 17 00:00:00 2001
From: Oswald Buddenhagen <oswald.buddenhagen@digia.com>
Date: Wed, 12 Dec 2012 12:16:09 +0100
Subject: [PATCH] fix installation of aux headers

the source paths need to be made relative to the main source dir

Change-Id: I98d74eca7e54d1179ab602bb87b96b66f3193ce4
Reviewed-by: Joerg Bornemann <joerg.bornemann@digia.com>
Reviewed-by: Oswald Buddenhagen <oswald.buddenhagen@digia.com>
---
 bin/syncqt | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/bin/syncqt b/bin/syncqt
index ff55a9a..ac81b0a 100755
--- a/bin/syncqt
+++ b/bin/syncqt
@@ -964,7 +964,7 @@ foreach my $lib (@modules_to_sync) {
                             if($public_header) {
                                 #deal with the install directives
                                 if($public_header) {
-                                    my $pri_install_iheader = fixPaths($iheader, $current_dir);
+                                    my $pri_install_iheader = fixPaths($iheader, $dir);
                                     foreach my $class (@classes) {
                                         # Strip namespaces:
                                         $class =~ s/^.*:://;
@@ -972,23 +972,24 @@ foreach my $lib (@modules_to_sync) {
     #                                       $class =~ s,::,/,g;
     #                                   }
                                         my $class_header = fixPaths("$out_basedir/include/$lib/$class",
-                                                                    $current_dir) . " ";
+                                                                    $dir) . " ";
                                         $pri_install_classes .= $class_header
                                                                     unless($pri_install_classes =~ $class_header);
                                     }
                                     if ($explicitheaders{$lib}{basename($iheader)}) {
-                                        my $compat_header = fixPaths("$out_basedir/include/$lib/$explicitheaders{$lib}{basename($iheader)}", $current_dir) . " ";
+                                        my $compat_header = fixPaths("$out_basedir/include/$lib/$explicitheaders{$lib}{basename($iheader)}",
+                                                                     $dir) . " ";
                                         $pri_install_files .= $compat_header unless($pri_install_files =~ $compat_header);
                                     }
                                     $pri_install_files.= "$pri_install_iheader ";;
                                 }
                             }
                             elsif ($qpa_header) {
-                                my $pri_install_iheader = fixPaths($iheader, $current_dir);
+                                my $pri_install_iheader = fixPaths($iheader, $dir);
                                 $pri_install_qpafiles.= "$pri_install_iheader ";;
                             }
                             else {
-                                my $pri_install_iheader = fixPaths($iheader, $current_dir);
+                                my $pri_install_iheader = fixPaths($iheader, $dir);
                                 $pri_install_pfiles.= "$pri_install_iheader ";;
                             }
                         }
-- 
1.8.0.2

